rotkehlchen/api/rest.py

```diff
@@ -4125,3 +4120,111 @@ def get_all_counterparties(self) -> Response:
                 'message': '',
             },
         )
+
+    def get_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            addresses: Optional[List[ChecksumEthAddress]],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        entries = db_addressbook.get_addressbook_entries(
+            book_type=book_type,
+            addresses=addresses,
+        )
+        serialized = [entry.serialize() for entry in entries]
+        return api_response(_wrap_in_ok_result(serialized))
+
+    def add_addressbook_entries(
+            self,
+            book_type: AddressbookType,
+            entries: List[AddressbookEntry],
+    ) -> Response:
+        db_addressbook = DBAddressbook(self.rotkehlchen.data.db)
+        try:
+            db_addressbook.add_addressbook_entries(book_type=book_type, entries=entries)
+            return api_response(result=OK_RESULT)
+        except InputError as e:
+            return api_response(
+                result=wrap_in_fail_result(str(e)),
+                status_code=HTTPStatus.CONFLICT,
+            )
```

Comment:
One thing that I noticed is that you can have addresses of the user with names in the global book. I believe it shouldn't be a problem unless you start using names like `my cold storage address` where you can clearly identify uses addresses. I would disallow to add users addresses to the global book. Opinions around this accepted

Reply:
The initial idea of the address book (if I recall correctly) was to have 2 address books. And global address book is intended to be a place where users can add publicly known addresses that are not provided by our app. Also the benefit of the global address book is that it allows to access mappings from any of the user's account. And in my opinion it is okay to have global address book if we properly notify users in frontend about how it works. Also users will always have a choice and can simply don't use global address book at all if they wish.
 What do you think?

Reply:
I'm not sure whether you understand it so will explain. The PR introduces 2 address books. One - global (that we were discussing here). And one private - it is stored in the encrypted database. And a user can choose for each mapping where to store it - in global db or in private. Does it make sense?

If it's not clear let me know so I could improve API documentation.

Reply:
Yes, that I understood. The issue I see is that if any of the addresses that wants to be named is owned by the user it shouldn't be in any way stored in the non encrypted database. And if the backend detects that this wants to be done should be prevented.

Reply:
Your idea is cool but how do you see it to be implemented? I mean, is there a way to programmatically check wether an address is a publicly known address or a user's address?

Reply:
Ah..... You mean this thing.... First a user probably won't be trying to add their address to an address book since all addresses have their own label . Second, are you sure that we should force stop a user to add an address to the address book? I mean it's dangerous if a private address gets exposed but it is something that frontend should notify a user about since it is not a thing that we want to completely prohibit. Or am I wrong?

Reply:
Yes, exactly. I see the path where a user sees that one of his account is not labeled and instead of labeling it as (s)he should he goes to the global naming and tries to add a label here, and you can't trust the frontend for that, you should validate against it and protect one user from this. But we can get a third opinion from @LefterisJP  or @prettyirrelevant

Reply:
It's all about presentation. We should explain that the global address book is public and can be even (in the future) shared with others.

Reply:
Agree with you @LefterisJP !